### ISSUE TITLE
LibWeb/CSS: Properties that don't support `calc()` but should

### ISSUE THREAD

**Issue by AtkinsSJ** [LadybirdBrowser/ladybird#1812]
Any property that allows a numeric type (either a `<number>`, `<integer>`, or a number with a unit like a `<length>` or `<percentage>` or any other dimension type) can also be a math function such as `calc()`. These require special handling in the code to store the calculation, and resolve them when needed.

Generally:
- If the property has bespoke parsing code, that code needs to make sure that any parsing functions it uses also handle math functions. (The methods that parse a numeric type as a `CSSStyleValue` all do this already.)
- The property's value in `StyleProperties.h` needs to be a type that can store a math function - for example, for `Length` that would be `LengthOrCalculated`. Note that the `*Percentage` types can also store math functions.
- The `StyleProperties` getter for the property, if it has one, needs to also check for math values and resolve them. That will likely require passing in a `Layout::Node`.

The following properties have been identified by the `Tests/LibWeb/Text/input/css/calc-coverage.html` test added in #1788. Others might pass this test but not handle math functions later on in the process - I'll try and add those when I spot them.

- [ ] grid-auto-columns
- [ ] grid-auto-rows
- [ ] grid-template-columns
- [ ] grid-template-rows
- [ ] transform-origin

Please rebaseline `Tests/LibWeb/Text/input/css/calc-coverage.html` when fixing any of these!

**Comment by ankushChatterjee** [comment]
I have created PR: #1934 for transform-origin

**Comment by F3n67u** [comment]
`calc()` support filter and backdrop-filter was added by https://github.com/LadybirdBrowser/ladybird/pull/2037

**Comment by Heisen47** [comment]
hi , is this issue solved?

**Comment by nhlmg93** [comment]
Anyway we could get a refresh of this issue? some of the files referenced here no longer exist and were potentially broken up/refactored. makes it hard to follow as a first time contributor and there are not many issues tagged. just really trying to find a good jumping on point.

@AtkinsSJ

**Comment by AtkinsSJ** [comment]
I've updated it. `StyleProperties` got renamed to `ComputedProperties` at some point.

This does still need work, but the remaining properties, all `grid-` related, are a bit tricky, so I'm removing the **good first issue** label.
